src/coreclr/vm/methodtablebuilder.cpp

```diff
@@ -1694,6 +1694,45 @@ MethodTableBuilder::BuildMethodTableThrowing(
         &pByValueClassCache, bmtMFDescs, bmtFP,
         &totalDeclaredFieldSize);
 
+    if (IsValueClass())
+    {
+        const void* pVal;                  // The custom value.
+        ULONG       cbVal;                 // Size of the custom value.
+        HRESULT hr = GetCustomAttribute(bmtInternal->pType->GetTypeDefToken(),
+            WellKnownAttribute::InlineArrayAttribute,
+            &pVal, &cbVal);
+
+        if (hr != S_FALSE)
+        {
+            if (bmtEnumFields->dwNumInstanceFields != 1)
+            {
+                // TODO: diagnostics, must have one field.
```

Comment:
is this a spurious TODO?

Reply:
This is the same TODO as in CorCLR - about friendly error message. I will not merge with TODOs